### DISCUSSION TITLE
Jit Error

### DISCUSSION THREAD

**Discussion by rtapia8484** [casadi/casadi#3854] [Q&A]
I have been running the example nlp_codegen.py to understand the jit feature, I am running Windows but have installed gcc and while the mode="external" does work, the mode="jit" options give me the following error:

CasADi - 2024-09-16 14:42:54 MESSAGE("calling "gcc -O3 /c jit_tmpkljxq6.c /Fotmp_casadi_compiler_shellkzpdis.obj"") [.../casadi/solvers/shell_compiler.cpp:229]
CasADi - 2024-09-16 14:42:54 WARNING("Failed to remove tmp_casadi_compiler_shellkzpdis.dll") [.../casadi/solvers/shell_compiler.cpp:66]
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
<ipython-input-13-a26faba72764> in ?()
     57     # jit_options = {"verbose": True, "compiler": compiler}
     58     options = {"jit": True, "compiler": "shell", "jit_options": jit_options}
     59 
     60     # Create an NLP solver instance
---> 61     solver = nlpsol("solver", "ipopt", nlp, options)

~\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\LocalCache\local-packages\Python312\site-packages\casadi\casadi.py in ?(*args)
  29374 
  29375 
  29376 
  29377     """
> 29378     return _casadi.nlpsol(*args)

RuntimeError: .../casadi/core/function_internal.cpp:154: Error calling IpoptInterface::finalize for 'solver':
.../casadi/solvers/shell_compiler.cpp:231: Compilation failed. Tried "gcc -O3 /c jit_tmpkljxq6.c /Fotmp_casadi_compiler_shellkzpdis.obj"

I am aware of the FAQ jit post, but I don't see how it applies.

Thanks,

**Comment by jgillis** [comment]
Hi,
If we look in the source we can see that visual-studio style flags are by default enabled on Windows:
https://github.com/casadi/casadi/blob/81bbcd37d9aa69e53cae7164e5c5c06c5f8529ee/casadi/solvers/shell_compiler.cpp#L146-L161

A solution is to override those defaults with gcc-style flags by adding more entries to `jit_options`
